### ISSUE TITLE
[BUG] Conda package is missing clustering.deep_learning module

### ISSUE THREAD

**Issue by benjaminh** [aeon-toolkit/aeon#986]
### Describe the bug

After a fresh install of aeon 0.5.0 in a virtual environment with conda, I got `ModuleNotFoundError: No module named 'aeon.clustering.deep_learning'`

### Steps/Code to reproduce the bug

- create and activate a virtual environment with conda : `conda create -n aeon-env` and `conda activate aeon-env`
- install aeon : `conda install -c conda-forge aeon`
- check aeon 0.5.0 is installed : `conda list`
- try to import `AEFCNClusterer` from `aeon.clustering.deep_learning`

Besides, one can confirm the `deep_learning` module is actually missing from the conda package by looking in the installation repository within the newly created virtual environment.

### Expected results

No error shown.

### Actual results

Missing `deep_learning` module in `aeon.clustering` in python package, and `ModuleNotFoundError` raised when trying to load `AEFCNClusterer` from `aeon.clustering.deep_learning` module.

### Versions

- aeon v0.5.0 (aeon 0.5.0  pyhd8ed1ab_0  conda-forge)
- conda 23.10.0

**Comment by TonyBagnall** [comment]
thanks for highlighting this @benjaminh we will take a look

**Comment by MatthewMiddlehurst** [comment]
Hello, sorry for the delay on this. `AEFCNClusterer` was not in release 0.5.0, but our documentation was being updated to reflect the `main` branch instead of the latest release. This is our bad, I have swapped the default webpage to reflect the latest release now to prevent confusion in the future.

 `AEFCNClusterer` is in the recently released v0.6.0, however. To double-check the 0.6.0 conda, I ran the following code after using `conda install -c conda-forge aeon`.

```
from aeon.clustering.deep_learning import AEFCNClusterer
from aeon.datasets import load_arrow_head

if __name__ == "__main__":
    X, y = load_arrow_head()
    clusterer = AEFCNClusterer(3)
    clusterer.fit(X)
    print(clusterer.predict(X))
```

...and got a soft dependency error 🙂. After also using `conda install -c conda-forge tensorflow`, It started running as expected, though.
